frontend/components/UpgradeURD.tsx

```diff
@@ -0,0 +1,315 @@
+import React, { useContext, useEffect, useState } from 'react';
+import {
+  Box,
+  Button,
+  Flex,
+  Step,
+  StepDescription,
+  StepIndicator,
+  StepNumber,
+  Stepper,
+  StepSeparator,
+  StepStatus,
+  StepTitle,
+  Text,
+  useColorModeValue,
+  useSteps,
+  useToast,
+  VStack,
+} from '@chakra-ui/react';
+import { WalletContext } from '@/components/wallet/WalletContext';
+import { ethers } from 'ethers';
+import LSP1GraveForwarderAbi from '@/abis/LSP1GraveForwarder.json';
+import { LSP1GraveForwarder } from '@/contracts';
+import { formatAddress } from '@/utils/tokenUtils';
+import { FaCheckCircle } from 'react-icons/fa';
+import {
+  resetLSPDelegates,
+  setForwarderAsLSPDelegate,
+  updateBECPermissions,
+} from '@/utils/urdUtils';
+
+const initialLeavingSteps = [
+  {
+    title: 'Give your 🆙 necessary permissions',
+    instructions: 'Give permissions to your Browser Extension Controller.',
+    instructions2: { text: 'This can also be done manually.', address: null },
+    completeText: { text: 'PERMISSION SET', address: null },
+    complete: false,
+  },
+  {
+    title: 'Reset previous LSP 7 and LSP8 delegate permissions',
+    completeText: { text: '', address: null },
+    complete: false,
+  },
+  {
+    title: 'Direct all 🆙 spam to the GRAVE',
+    completeText: { text: 'SPAM IS DEAD', address: null },
+    complete: false,
+  },
+  {
+    title: 'Migrate your GRAVE to the new forwarder',
```

Comment:
migration step needs to happen before setting new forwarder as LSP delegate. the reason is to avoid a broken state for the full flow. if the user executes steps 1-3, but not 4 they're in a broken state. meaning if they get an asset sent to their UP the transaction will revert because no grave vault exists associated to new forwarder. we want to direct spam to GRAVE only once we've set up everything up to work. @democrazyfi

Reply:
Yeah I had them the other way but the reason why I moved them is because when we setUrd7/8 the view changes state to the subscribed view

But I'll move  those bits to the end of the flow so that doesn't happen